internal/scheduleringester/schedulerdb.go

```diff
@@ -289,3 +317,17 @@ func execBatch(ctx *armadacontext.Context, tx pgx.Tx, batch *pgx.Batch) error {
 	}
 	return nil
 }
+
+func multiColumnRunsUpdateStmt(id, phaseColumn, timeStampColumn string) string {
+	idPgType := "uuid"
+	if id == "job_id" {
+		idPgType = "text"
+	}
+	return fmt.Sprintf(`update runs set
+	%[2]v = runs_temp.%[2]v,
+	%[3]v = runs_temp.%[3]v
+	from (select * from unnest($1::%[4]v[], $2::boolean[] ,$3::timestamptz[])) 
+	as runs_temp(%[1]v, %[2]v, %[3]v) 
+	where runs.%[1]v = runs_temp.%[1]v;`,
+		id, phaseColumn, timeStampColumn, idPgType)
+}
```

Comment:
`unnest` seems more suitable here, much more performant and cleaner ( since we directly pass the array of vals into pg instead of destructing it on the `tx.Exec()` function)